docs/infrastructure/vault.md

```diff
@@ -0,0 +1,57 @@
+
+## Background
+Vault is a project from [Hashicorp](https://www.hashicorp.com/) for centralized, secure and auditable secrets management. This is useful from an operational context because it removes the need for tracking sensitive information in source control or using cobbled together solutions. Vault can provide a single location to store and generate passwords, API keys, SSL certificates, etc. While the installation and configuration of Vault is fairly painless, the difficulty arises in bringing a newly deployed Vault instance or cluster to a point where there is an authenticated client that can read or write data.
+
+The reason that an end-to-end setup is difficult is due to the fact that before a Vault node can be used it needs to be [initialized and unsealed](https://www.vaultproject.io/intro/getting-started/deploy.html). To do this in a fully automated manner while maintaining an appropriate level of security is non-trivial. To make this possible some custom integration work was needed. The relevant pieces that were required are contained in the following locations:
+- https://github.com/mitodl/hvac
+- https://github.com/mitodl/salt-extensions/blob/master/extensions/_modules/vault.py
+- https://github.com/mitodl/salt-extensions/blob/master/extensions/_states/vault.py
+- https://github.com/mitodl/vault-formula
+
+## Workflow
+The process by which a newly created instance or cluster can be brought under the control of a Salt master is as follows:
+
+1. One or more server instances are provisioned (e.g. an EC2 node)
+2. Vault (and, optionally, Consul) is installed to the server
+3. Salt initializes (one of) the Vault instance(s)
+
+This is the point at which the automated process is generally handed to an operator to complete the setup manually with unsealing the vault and provisioning authentication backends. In our setup the following steps can optionally be performed in a fully automated manner:
+
+4. Unseal the initialized vault
+5. (optional) Regenerate the sealing keys and encrypt them with the operators public GPG key(s)
+6. Provision the EC2 authentication backend
+7. Create a Vault policy with an appropriate level of access
+8. Create an EC2 instance profile that links the policy with an appropriate subset of the data needed to identify a salt master (e.g. IAM instance profile and AWS account number)
+9. Authenticate the salt master using the generated EC2 instance profile and write the returned client token to a configuration file on the master instance
+
+## How It Works
```

Comment:
If I had to pick something to comment on, I would say that I had to read this a few times to get it. Might be just me, but guess it might also help rewording it.